spec/controllers/boxes_controller_spec.rb

```diff
@@ -109,64 +117,103 @@
     end
 
     it "should be accessible to institution owner" do
-      get :print, params: { id: box.id }
+      get :print, params: { id: @floating_box.id }
       expect(response).to have_http_status(:ok)
     end
 
     it "should be allowed if can read" do
       grant user, other_user, box, READ_BOX
       sign_in other_user
 
-      get :print, params: { id: box.id }
+      get :print, params: { id: @confirmed_box.id, context: other_institution.uuid }
       expect(response).to have_http_status(:ok)
     end
 
     it "shouldn't be allowed if can't read" do
       sign_in other_user
+      get :print, params: { id: @floating_box.id, context: other_institution.uuid }
+      expect(response).to have_http_status(:forbidden)
+    end
 
-      get :print, params: { id: box.id }
+    it "confirmed box should be accessible to sender" do
+      get :print, params: { id: @confirmed_box.id }
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "floating box shouldn't be accessible receiver" do
+      sign_in other_user
+      get :print, params: { id: @floating_box.id, context: other_institution.uuid }
       expect(response).to have_http_status(:forbidden)
     end
   end
 
   describe "inventory" do
     it "should be accessible to institution owner" do
-      get :inventory, params: { id: box.id, format: "csv" }
+      get :inventory, params: { id: @floating_box.id, format: "csv" }
       expect(response).to have_http_status(:ok)
       expect(response.content_type).to eq("text/csv")
-      expect(response.body.strip.split("\n").size).to eq(box.samples.count + 1)
+      expect(response.body.strip.split("\n").size).to eq(@floating_box.samples.count + 1)
       expect(response.body).to_not match("Blinded")
-      expect(response.headers["Content-Disposition"]).to match(/cdx_box_inventory_#{box.uuid}\.csv/)
+      expect(response.headers["Content-Disposition"]).to match(/cdx_box_inventory_#{floating_box.uuid}\.csv/)
+    end
+
+    it "should be ordered by batch_number, concentration, replicate ASC" do
+      get :inventory, params: { id: @confirmed_box.id, format: "csv" }
+      expect(response).to have_http_status(:ok)
+      @as_its = []
+      CSV.parse(response.body).tap(&:shift).each do |row|
+        @as_its << { :batch_number => row[3], :concentration => row[6], :replicate => row[7] }
+      end
+      @sorted = @as_its
+      expect(@as_its).to eq(@sorted.sort_by{ |sample|  [ sample[:batch_number], sample[:concentration], sample[:replicate] ] })
```

Comment:
Nit picking:

1. you don't need to use instance variables (`@as_its` and `@sorted` are locals).
2. you can leverage map to improve readability:

```ruby
results = CSV.parse().tap(&:shift).map do |row|
  { batch_number: ... }
end
```